cli/src/generate/prepare_grammar/expand_tokens.rs

```diff
@@ -139,8 +141,11 @@ pub(crate) fn expand_tokens(mut grammar: ExtractedLexicalGrammar) -> Result<Lexi
 impl NfaBuilder {
     fn expand_rule(&mut self, rule: &Rule, mut next_state_id: u32) -> Result<bool> {
         match rule {
-            Rule::Pattern(s) => {
+            Rule::Pattern(s, f) => {
                 let s = preprocess_regex(s);
+                if f.contains('i') {
+                    self.case_insensitive = true;
+                }
```

Comment:
May be it's better to pass flags directly to `expand_regex` and don't polute the self namespace?

Reply:
I thought about this before (and initially did it this way), it'd pollute expand_regex but that's fine

Reply:
No, I meant it irrelevant for the self namespace because it's only used in expand regex so it's better to pass whole flags as additional parameter. It's not good idea to store everything in the self.